File: elixir/hanoi_tower.ex
defmodule HanoiTower do
  def hanoi(n), do: hanoi(n, 1, 2, 3)
  def hanoi(1, _, _, _), do: 1
  def hanoi(n, a, b, c), do: hanoi(n-1, a, c, b) + 1 + hanoi(n-1, b, a, c)
end

# moves = HanoiTower.hanoi(32,"a","b","c")
# IO.puts("#{moves} moves")

File: erlang/hanoi_tower.erl
-module (hanoi_tower).
-export([start/0]).

hanoi(1,_A,_B,_C) -> 1 ;
hanoi(N,A,B,C) -> hanoi(N-1,A,C,B) + 1 + hanoi(N-1,B,A,C) .

start() ->
  Moves = hanoi(32,'a','b','c') ,
  io:fwrite("Moves: ~B~n", [Moves]) .
